### ISSUE TITLE
files are re-downloaded after fixing hash

### ISSUE THREAD

**Issue by 2-www** [flatpak/flatpak-builder#604]
### Checklist

- [X] I agree to follow the [Code of Conduct](https://github.com/flatpak/flatpak/blob/master/CODE_OF_CONDUCT.md) that this project adheres to.
- [X] I have searched the [issue tracker](https://www.github.com/flatpak/flatpak-builder/issues) for a bug that matches the one I want to file, without success.

### flatpak-builder version

flatpak-builder 1.4.1

### Flatpak version

Flatpak 1.14.8

### How to reproduce

1. write a module definition that has the `"sha256"` property empty or wrong
2. see error: `Failed to download sources: module example-module: Wrong sha256 checksum for file.tar.gz, expected "...", was "..."`
3. fix the sha256 in the manifest

### Expected Behavior

uses already downloaded file

### Actual Behavior

downloads huge file again

**Comment by nanonyme** [comment]
I think it makes sense file is not preserved on wrong hash. It's typically corrupted.

**Comment by TingPing** [comment]
I agree with @nanonyme, it's best to keep known good data in the cache.
